src/shared_modules/indexer_connector/tests/unit/serverSelector_test.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * Wazuh Indexer Connector - ServerSelector tests
+ * Copyright (C) 2015, Wazuh Inc.
+ * September 08, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "serverSelector_test.hpp"
+#include "serverSelector.hpp"
+#include <chrono>
+#include <memory>
+#include <string>
+#include <thread>
+
+/**
+ * @brief Test instantiation with valid servers.
+ *
+ */
+TEST_F(ServerSelectorTest, TestInstantiation)
+{
+    EXPECT_NO_THROW(m_selector = std::make_shared<ServerSelector>(m_servers));
+}
+
+/**
+ * @brief Test instantiation without servers.
+ *
+ */
+TEST_F(ServerSelectorTest, TestInstantiationWithoutServers)
+{
+    m_servers.clear();
+
+    // It doesn't throw an exception because the class ServerSelector accepts vector without servers
+    EXPECT_NO_THROW(m_selector = std::make_shared<ServerSelector>(m_servers));
+}
+
+/**
+ * @brief Test instantiation and getNext server before health check.
+ *
+ */
+TEST_F(ServerSelectorTest, TestGetNextBeforeHealthCheck)
+{
+    const auto hostGreenServer {m_servers.at(0)};
+    const auto hostRedServer {m_servers.at(1)};
+
+    std::string nextGreenServer;
+    std::string nextRedServer;
+
+    EXPECT_NO_THROW(m_selector = std::make_shared<ServerSelector>(m_servers));
+
+    // It doesn't thrown and exception because all servers are available before health check
+    EXPECT_NO_THROW(nextGreenServer = m_selector->getNext());
+    EXPECT_EQ(nextGreenServer, hostGreenServer);
+
+    // It doesn't thrown and exception because all servers are available before health check
+    EXPECT_NO_THROW(nextRedServer = m_selector->getNext());
+    EXPECT_EQ(nextRedServer, hostRedServer);
+}
+
+/**
+ * @brief Test instantiation and getNext server before and after health check.
+ *
+ */
+TEST_F(ServerSelectorTest, TestGetNextBeforeAndAfterHealthCheck)
+{
+    const auto hostGreenServer {m_servers.at(0)};
+    const auto hostRedServer {m_servers.at(1)};
+
+    std::string nextGreenServer;
+    std::string nextRedServer;
+
+    EXPECT_NO_THROW(m_selector = std::make_shared<ServerSelector>(m_servers));
+
+    // It doesn't thrown and exception because all servers are available before health check
+    EXPECT_NO_THROW(nextGreenServer = m_selector->getNext());
+    EXPECT_EQ(nextGreenServer, hostGreenServer);
+
+    // It doesn't thrown and exception because all servers are available before health check
+    EXPECT_NO_THROW(nextRedServer = m_selector->getNext());
+    EXPECT_EQ(nextRedServer, hostRedServer);
+
+    // Interval to check the health of the servers
+    std::this_thread::sleep_for(std::chrono::seconds(INTERVAL + 5));
```

Comment:
**Nitpick**: how about declaring a new variable with INTERVAL so we can give it another name to make it more clear, and may be it will be easier to change in the future when the interval is configurable?
Something like this:
`    constexpr auto SERVER_HEALTH_CHECK_INTERVAL = INTERVAL;`
and then:
`    std::this_thread::sleep_for(std::chrono::seconds(SERVER_HEALTH_CHECK_INTERVAL + 5));`

Reply:
I missed that, but yes.

Reply:
done